packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1479,35 +1482,148 @@ export class TransactionController extends BaseController<
+      transactionMeta,
+      pickBy({ hash, status }),
+    );
 
     if (status === TransactionStatus.submitted) {
-      transactionMeta.submittedTime = new Date().getTime();
-      transactionMeta.status = status;
+      updatedTransactionMeta.submittedTime = new Date().getTime();
     }
 
     if (status === TransactionStatus.failed) {
-      transactionMeta = {
-        ...transactionMeta,
-        error: normalizeTxError(new Error(errorMessage)),
-        status: TransactionStatus.failed,
-      };
+      updatedTransactionMeta.error = normalizeTxError(new Error(errorMessage));
     }
 
-    if (custodyStatus) {
-      transactionMeta.custodyStatus = custodyStatus;
-    }
+    this.updateTransaction(
+      updatedTransactionMeta,
+      `TransactionController:updateCustodialTransaction - Custodial transaction updated`,
+    );
+  }
 
-    if (hash) {
-      transactionMeta.hash = hash;
+  /**
+   * Creates approvals for all unapproved transactions persisted.
+   */
+  initApprovals() {
+    const chainId = this.getChainId();
+    const unapprovedTxs = this.state.transactions.filter(
+      (transaction) =>
+        transaction.status === TransactionStatus.unapproved &&
+        transaction.chainId === chainId,
+    );
+
+    for (const txMeta of unapprovedTxs) {
+      this.processApproval(txMeta, {
+        shouldShowRequest: false,
+      }).catch((error) => {
+        if (error?.code === errorCodes.provider.userRejectedRequest) {
+          return;
+        }
+        console.error('Error during persisted transaction approval', error);
+      });
     }
+  }
 
-    this.updateTransaction(
-      transactionMeta,
-      `TransactionController:updateCustodialTransaction - Custodial transaction updated`,
+  /**
+   * Search transaction metadata for matching entries.
+   *
+   * @param opts - Options bag.
+   * @param opts.searchCriteria - An object containing values or functions for transaction properties to filter transactions with.
+   * @param opts.initialList - The transactions to search. Defaults to the current state.
+   * @param opts.filterToCurrentNetwork - Whether to filter the results to the current network. Defaults to true.
+   * @param opts.limit - The maximum number of transactions to return. No limit by default.
+   * @returns An array of transactions matching the provided options.
+   */
+  getTransactions({
+    searchCriteria = {},
+    initialList,
+    filterToCurrentNetwork = true,
+    limit,
+  }: {
+    searchCriteria?: any;
+    initialList?: TransactionMeta[];
+    filterToCurrentNetwork?: boolean;
+    limit?: number;
+  } = {}) {
+    const chainId = this.getChainId();
+    // searchCriteria is an object that might have values that aren't predicate
+    // methods. When providing any other value type (string, number, etc), we
+    // consider this shorthand for "check the value at key for strict equality
+    // with the provided value". To conform this object to be only methods, we
```

Comment:
Given there info in this comment that describes the parameter `searchCriteria`, you could add more info to the tsdoc for the parameter above.

Reply:
I've copied the implementation almost exactly from the extension to begin with for the sake of time and stability.

These comments may be useful for audit sake and developers, but I've tried to condense the API doc though I still mention the value or functions assigned to the properties.